src/cryptsetup/cryptsetup.c

```diff
@@ -1076,6 +1103,21 @@ static int attach_luks_or_plain_or_bitlk_by_tpm2(
                         if (r != -EAGAIN) /* EAGAIN means: no tpm2 chip found */
                                 return r;
                 } else {
+                        r = attach_luks2_by_tpm2(cd, name, flags);
+                        /* EAGAIN  means: no tpm2 chip found
+                         * ENOTSUP means: no libcryptsetup plugins support */
+                        if (!IN_SET(r, -ENOTSUP, -EAGAIN)) {
```

Comment:
ENOTSUP is a weird error, as on the kernel side it's different from EOPNOTSUPP, but in userspace it's the same, and the latter is the primary name. In our codebase we unify on the name EOPNOTSUPP for this.